Marketing your package blog (#732)
ropensci/roweb3: content/blog/2024-02-23-package-marketing/index.md

---
title: Spread the word about your package
author:
- Yanina Bellini Saibene

> - Maelle Salmon

Reviewer: ```suggestion
- Maëlle Salmon
```
:nail_care: (I know I have a better keyboard :wink:)

Author: I feel an ache every time I write your name wrong because I can't set up my keyboard correctly.

---
title: Spread the word about your package
author:
- Yanina Bellini Saibene
- Maëlle Salmon
date: '2024-02-23'
slug: package-marketing
tags:
- community

description: ""
---

> Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools to let the people your package can help know that it is available. 

Reviewer: ```suggestion
Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools to let the people that your package can help, know that it is available. 
```

easier to parse

## Publish

> Create a comprehensive README that clearly explains what your package does, how to install it, how can user start using it. Include examples or use cases and any relevant information (or links to) that can help users understand how your package can help them. 

Reviewer: ```suggestion
Create a comprehensive README that clearly explains what your package does, how to install it, how users can start using it. Include examples or use cases and any relevant information (or links to) that can help users understand how your package can help them. 
```

> Create a universe on R-universe, which will help you by simplifying the installation of your package and provide with useful stats and information about it.  You can also publish it on CRAN or submit it to our peer review process. All this steps will support you with discoverability and also to comply with quality standars.  

Reviewer: ```suggestion
Create a universe on R-universe, which will help users by simplifying the installation of your package and provide with useful stats and information about it.  You can also publish it on CRAN or submit it to our peer review process. All this steps will support you with discoverability and also to comply with quality standards.  
```

Add that the universe is a way to showcase one's collection of packages, like GitHub pinned repos?

## Promote 

> For example, things maintainers can do:

Reviewer: write a tech note for rOpenSci blog?

Reviewer: even if it's in the following section.

---
title: Spread the word about your package
author:
- Yanina Bellini Saibene
- Maëlle Salmon
date: '2024-02-23'
slug: package-marketing
tags:
- community

description: "Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools to advertize your package."
---

> Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools to let the people that your package can help, know that it is available. 

Author: The whys I consider important are in the text (I can try to made them more clear):  
- "presenting it to the world is a crucial step to **gain visibility and attract users.**"
-  "to let the people that **your package can help know that it is available**. " 

But, other reason are completely valid too.  Should we list more options in the intro?

> Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools to let the people that your package can help, know that it is available. 

Reviewer 2: I guess my question was not so much why promote, but *why do you want to attract users* in the first place? I mean, users just mean more people bugging you with their problems, right? 🤣  Why do you even need users if they're not buying the product? What's the benefit? 

That was the gist of my thought process. So I would say, add one reason to qualify why you would want to get more users.

> Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools to let the people that your package can help, know that it is available. 

Author: To reach the people you package can help 😀.

> Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools to let the people that your package can help, know that it is available. 

Author: I added some more info for this point. Thank you! 🙏

## Publish

> Create a comprehensive README that clearly explains what your package does, how to install it, how users can start using it. Include examples or use cases and any relevant information (or links to) that can help users understand how your package can help them. 

Reviewer: ```suggestion
Create a comprehensive [README](https://blog.r-hub.io/2019/12/03/readmes/) that clearly explains what your package does, how to install it, how users can start using it. Include examples or use cases and any relevant information (or links to) that can help users understand how your package can help them. 
```

> [Create a universe on R-universe](/blog/2021/06/22/setup-runiverse/), which will help users by simplifying the installation of your package and provide with useful stats and information about it. R-Universe is also a way to showcase your packages collection. You can also [publish it on CRAN](https://cran.r-project.org) or submit it to our [peer review process](https://ropensci.org/software-review/). All this steps will support you with discoverability and also to comply with quality standards.  

Reviewer: ```suggestion
[Create a universe on R-universe](/blog/2021/06/22/setup-runiverse/), which will help users by simplifying the installation of your package, and which will provide with useful stats and information about it. R-Universe is also a way to showcase your packages collection. You can also submit it to our [peer review process](https://ropensci.org/software-review/) if in scope. You can [publish it on CRAN](https://cran.r-project.org). All these steps will support you with discoverability and also to comply with quality standards.  
```

## Promote 

Leverage platforms like Mastodon, LinkedIn, and R-specific forums such as [R-bloggers](https://www.r-bloggers.com) and [R Weekly](https://rweekly.org) to announce the release of your package. 

> > If you package is part of the rOpenSci Suite you can [write a blog post or a tech note](https://blogguide.ropensci.org/) for our blog and we will share in our [monthly newsletter](/news/) and our social media.

Reviewer: why use a quote formatting?

Author: I wasn't sure on how to highlight the rOpenSci support, I'm not convinced on using quote formatting.

> > If you package is part of the rOpenSci Suite you can [write a blog post or a tech note](https://blogguide.ropensci.org/) for our blog and we will share in our [monthly newsletter](/news/) and our social media.

Author: ```suggestion
If you package is part of the rOpenSci Suite you can [write a blog post or a tech note](https://blogguide.ropensci.org/) for our blog and we will share in our [monthly newsletter](/news/) and our social media.
```

> You can also share updates, tutorials, use cases, talks and events where you will participate to engage with users. 

Reviewer: share those where?

Author: Oh, on social media! I will add that.

> You can also share updates, tutorials, use cases, talks and events where you will participate to engage with users. 

Author: ```suggestion
You can also use your social media and your blog (if you have one) to share updates, tutorials, use cases, talks, and events where you will participate to engage with users. 
```

> > For rOpenSci packages, we collect the use cases and feature them on the website as well as the package. We also have social media campaing, for example "A Package a Day" on Mastodon were we feature one package of our suite each day. For LinkedIn we publish every Monday our "Package Weekly Digest" with a list of five package that belong to the same category.

Reviewer: ```suggestion
> For rOpenSci packages, we collect the [use cases](/use-cases) and feature them on the website as well as the package. We also have social media campaigns, for example "A Package a Day" on Mastodon were we feature one package of our suite each day. On LinkedIn we publish every Monday our "Package Weekly Digest" with a list of five package that belong to the same category.
```

> If you like to give talks, you can speak at an R User Group or R-Ladies Chapter, organizers invite spekaers but most of the groups are open to people that offer to give a talk or teach a workshop. Present your package on a field specific conference is another way to reach your users and get greater exposure.  There are R specific conferences like useR!, LatinR, satRdays and Posit::Conf. 

Reviewer: ```suggestion
If you like to give talks, you can speak at an R User Group or R-Ladies Chapter, organizers invite speakers but most of the groups are open to people that offer to give a talk or teach a workshop. Present your package on a field specific conference is another way to reach your users and get greater exposure.  There are R specific conferences like useR!, LatinR, satRdays and Posit::Conf. 
```

add links to the conference?

> > rOpenSci have two hours [co-working sessions each month](/coworking/), you can host a co-working session on the topic of the package.

Author: ```suggestion
rOpenSci have two hours [co-working sessions each month](/coworking/), you can host a co-working session on the topic of the package.
```

> > You can also propose an [rOpenSci Community Call](/commcalls/) relate with your package. For example, this community call about [targets](/commcalls/jan2023-targets/) has a panel with two users showing how they use the package and the principal author sharing new insights about the package.

Reviewer: I really think it'd be best not to use quote ">". Maybe instead split this section into "general advice" and "advice specific to rOpenSci packages"?

> > You can also propose an [rOpenSci Community Call](/commcalls/) relate with your package. For example, this community call about [targets](/commcalls/jan2023-targets/) has a panel with two users showing how they use the package and the principal author sharing new insights about the package.

Author: ```suggestion
You can also propose an [rOpenSci Community Call](/commcalls/) relate with your package. For example, this community call about [targets](/commcalls/jan2023-targets/) has a panel with two users showing how they use the package and the principal author sharing new insights about the package.
```

---
title: Spread the Word About Your Package
author:
- Yanina Bellini Saibene
- Maëlle Salmon
date: '2024-02-23'
slug: package-marketing
tags:
- community
- maintenance
- package-development

description: "Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools for advertizing your package."
---

Now that you have created your package, presenting it to the world is a crucial step to **gain visibility and attract users**. In addition to reaching the people your package can help, communicating about your package can help you find people to assist you in maintaining and improving your package and learn about how people use it.

In this blog post we suggest a series of activities and tools for advertising your package and to reach the people your package can help.   

> ## Publish

Reviewer: Sorry not to have thought of this sooner, should we mention pinning repositories on one's GitHub account if using GitHub? https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/pinning-items-to-your-profile

Author: Good idea!

Leverage platforms like Mastodon, LinkedIn, and R-specific forums such as [R-bloggers](https://www.r-bloggers.com) and [R Weekly](https://rweekly.org) to announce the release of your package. You can also use your own social media or blog to share updates, tutorials, use cases, talks, and upcoming events where you will be available to engage with users. 

If your package is part of the rOpenSci Suite you can [write a blog post or a tech note](https://blogguide.ropensci.org/) for our blog and we will share in our [monthly newsletter](/news/) and our social media.

For rOpenSci packages, we collect the [use cases](/usecases) and feature them on the website as well as the package. We also have social media campaigns, for example ["A Package a Day"](https://hachyderm.io/@rOpenSci/tagged/APackageADay) on Mastodon were we feature one package of our suite each day. On LinkedIn we publish every Monday our ["Package Weekly Digest"](https://www.linkedin.com/feed/update/urn:li:activity:7170464303186108416) with a list of five package that belong to the same category.

> If you like to give talks, you can speak at an R User Group or [R-Ladies Chapter](https://www.meetup.com/pro/rladies/). Organizers invite speakers but most groups are open to offers to give a talk or teach a workshop. Presenting your package at a domain-specific conference is another way to reach users and gain greater exposure. There are also R-specific conferences like useR!, LatinR, satRdays and Posit::Conf. 

Reviewer: ```suggestion
If you like to give talks, you can speak at an R User Group or [R-Ladies Chapter](https://www.meetup.com/pro/rladies/). Organizers invite speakers but most groups are open to offers to give a talk or teach a workshop. Presenting your package at a domain-specific conference is another way to reach users and gain greater exposure. There are also R-specific conferences like [useR!](https://events.linuxfoundation.org/user/), [LatinR](https://latin-r.com/), [satRdays](https://satrdays.org/) and [Posit::Conf](https://posit.co/conference/). 
```

rOpenSci has two-hour [co-working sessions each month](/coworking/), feel free to contact us if you're interested in hosting a co-working session on a topic related to your package.

> If your package has a broad scope, you can also propose an [rOpenSci Community Call](/commcalls/) on your package. For example, a community call about [targets](/commcalls/jan2023-targets/) included a panel with two users showing how they use the package and the principal author sharing new insights about the package. Alternatively, consider getting together with other package authors and proposing a domain-related Community Call, covering the use of several, more specific packages (e.g., [Research Applications of rOpenSci Taxonomy and Biodiversity Tools](/commcalls/2019-03-27/)).

Reviewer: ```suggestion
If your package has a broad scope, you can also propose an [rOpenSci Community Call](/commcalls/) on your package. For example, a community call about [targets](/commcalls/jan2023-targets/) included a panel with two users showing how they use the package and the principal author sharing new insights about the package. Alternatively, consider getting together with other package authors and proposing a domain-related Community Call, covering the use of several, more specific packages (for instance, [Research Applications of rOpenSci Taxonomy and Biodiversity Tools](/commcalls/2019-03-27/)).
```

You can check if your package fits in a [CRAN Task View](https://cran.r-project.org/web/views/) and propose the addition of your package.

[Packages in the rOpenSci suite are featured on our webpage](/packages/) and we provide pkgdown documentation websites. 

> Publishing a paper about your package is an important aspect for academics developers. General R journals for publishing R packages are [The R Journal](https://journal.r-project.org) and the [Journal of Statistical Software](https://www.jstatsoft.org/index), but remember that domain-specific journals may also publish R packages as tools for the field (e.g., [Methods in Ecology and Evolution](https://besjournals.onlinelibrary.wiley.com/journal/2041210x))

Reviewer: ```suggestion
Publishing a paper about your package is an important aspect for academics developers. General R journals for publishing R packages are [The R Journal](https://journal.r-project.org) and the [Journal of Statistical Software](https://www.jstatsoft.org/index), but remember that domain-specific journals may also publish R packages as tools for the field (for instance, [Methods in Ecology and Evolution](https://besjournals.onlinelibrary.wiley.com/journal/2041210x))
```

> The package that are peer reviwed by rOpenSci have a fast track for publication on the [Journal Open Source Software](https://joss.theoj.org).

Reviewer: ```suggestion
The package that are peer-reviewed by rOpenSci have a fast track for publication on the [Journal Open Source Software](https://joss.theoj.org).
```

> 

Reviewer: ```suggestion
## Conclusion

```

---

> title: Spread the Word About Your Package

Author: ```suggestion
title: Marketing Ideas For Your Package
```

author:
- Yanina Bellini Saibene
- Maëlle Salmon

> date: '2024-02-23'

Author: ```suggestion
date: '2024-03-06'
```

slug: package-marketing
tags:
- community
- maintenance
- package-development

description: "Now that you have created your package, presenting it to the world is a crucial step to gain visibility and attract users.  In this blog post we suggest a series of activities and tools for advertizing your package."
---

> Now that you have created your package, presenting it to the world is a crucial step to **gain visibility and attract users**. In addition to reaching the people your package can support, communicating about your package can help you find people to assist you in maintaining and improving your package, and can help you learn about how people use it.

Author: ```suggestion
Now that you have created your package, presenting it to the world is a crucial step to **gain visibility and attract users**. 

Marketing your package effectively contributes to reaching the people your package can support, finding people to assist you in maintaining and improving your package and allowing you to learn about how people use it.
```

Author: @maelle what about now?

In this blog post we suggest a series of activities and tools for advertising your package and to reach the people your package can help.   

## Publish

> 

Author: ```suggestion
If you use GitHub, you can [pin your package repository to your profile](https://docs.github.com/en/account-and-profile/setting-up-and-managing-your-github-profile/customizing-your-profile/pinning-items-to-your-profile) so other people can find it quickly. 
```
